bdikit/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/cl_api.py

```diff
@@ -105,15 +114,58 @@ def _sample_to_15_rows(self, table: pd.DataFrame):
                 table = unique_rows.sample(n=15, random_state=1)
         return table
 
+    def _check_gdc_cache(self, table: pd.DataFrame):
+
+        gdc_df = pd.read_csv(GDC_TABLE_PATH)
+        gdc_hash = hash_dataframe(gdc_df)
```

Comment:
Can we move these two lines to the `utils.py` and store the variables in private global variables in that module to avoid re-reading and re-hashing the file every time we call it? 

Perhaps we should also move the whole function since it is probably general enough to be re-used in other classes/functions.